src/Sulu/Bundle/AdminBundle/Resources/js/components/BlockCollection/BlockCollection.js

```diff
@@ -40,15 +41,20 @@ class BlockCollection extends React.Component<Props> {
 
     fillArrays() {
         const {defaultType, onChange, minOccurs, value} = this.props;
-        const {expandedBlocks} = this;
+        const {expandedBlocks, generatedBlockIds} = this;
 
         if (!value) {
             return;
         }
 
         expandedBlocks.push(...new Array(value.length - expandedBlocks.length).fill(false));
+        generatedBlockIds.push(
+            ...new Array(value.length - generatedBlockIds.length).fill(false).map(() => ++BlockCollection.idCounter)
+        );
         if (minOccurs && value.length < minOccurs) {
             expandedBlocks.push(...new Array(minOccurs - value.length).fill(true));
+            generatedBlockIds.push(...new Array(minOccurs - value.length).map(() => ++BlockCollection.idCounter));
```

Comment:
i guess you need to add a `.fill(false)` here too then?

Reply:
Right, going to add it.